mne_bids/write.py

```diff
@@ -1095,6 +1104,11 @@ def write_raw_bids(raw, bids_path, events_data=None,
         raise RuntimeError('You passed event_id, but no events_data NumPy '
                            'array. You need to pass both, or neither.')
 
+    if format not in CONVERT_FORMATS:
+        raise ValueError(f'The "format" {format} is not an '
+                         f'accepted format. Please use one of '
+                         f'{CONVERT_FORMATS}')
```

Comment:
This check should come later after line 1286 IMHO ... for example if a user specifies `BrainVision` but has MEG data, we should raise an error as well.

Finally, the current warning is confusing: you are talking about "accepted format", which will remind users of accepted formats in BIDS ... but what **you** actually mean is accepted input parameters to the write_raw_bids function.

Imagine I pass "EDF" ... you tell me it's not an accepted format ... but I know that it is. --> the warning msg should be something like "not supported input, please use ... "

Reply:
This is fixed in the latest commit. Lmk wyt.